Mathlib/MeasureTheory/Function/LpSeminorm/CompareExp.lean

```diff
@@ -0,0 +1,268 @@
+#align measure_theory.snorm'_le_snorm_ess_sup MeasureTheory.snorm'_le_snormEssSup
+
+theorem snorm_le_snorm_of_exponent_le {p q : ℝ≥0∞} (hpq : p ≤ q) [IsProbabilityMeasure μ]
+    (hf : AEStronglyMeasurable f μ) : snorm f p μ ≤ snorm f q μ :=
+  (snorm_le_snorm_mul_rpow_measure_univ hpq hf).trans (le_of_eq (by simp [measure_univ]))
+#align measure_theory.snorm_le_snorm_of_exponent_le MeasureTheory.snorm_le_snorm_of_exponent_le
+
+theorem snorm'_lt_top_of_snorm'_lt_top_of_exponent_le {p q : ℝ} [IsFiniteMeasure μ]
+    (hf : AEStronglyMeasurable f μ) (hfq_lt_top : snorm' f q μ < ∞) (hp_nonneg : 0 ≤ p)
+    (hpq : p ≤ q) : snorm' f p μ < ∞ := by
+  rcases le_or_lt p 0 with hp_nonpos | hp_pos
+  · rw [le_antisymm hp_nonpos hp_nonneg]
+    simp
+  have hq_pos : 0 < q := lt_of_lt_of_le hp_pos hpq
+  calc
+    snorm' f p μ ≤ snorm' f q μ * μ Set.univ ^ (1 / p - 1 / q) :=
+      snorm'_le_snorm'_mul_rpow_measure_univ hp_pos hpq hf
+    _ < ∞ := by
+      rw [ENNReal.mul_lt_top_iff]
+      refine' Or.inl ⟨hfq_lt_top, ENNReal.rpow_lt_top_of_nonneg _ (measure_ne_top μ Set.univ)⟩
+      rwa [le_sub_comm, sub_zero, one_div, one_div, inv_le_inv hq_pos hp_pos]
+#align measure_theory.snorm'_lt_top_of_snorm'_lt_top_of_exponent_le MeasureTheory.snorm'_lt_top_of_snorm'_lt_top_of_exponent_le
+
+theorem Memℒp.memℒp_of_exponent_le {p q : ℝ≥0∞} [IsFiniteMeasure μ] {f : α → E} (hfq : Memℒp f q μ)
+    (hpq : p ≤ q) : Memℒp f p μ := by
+  cases' hfq with hfq_m hfq_lt_top
+  by_cases hp0 : p = 0
+  · rwa [hp0, memℒp_zero_iff_aestronglyMeasurable]
+  rw [← Ne.def] at hp0
+  refine' ⟨hfq_m, _⟩
+  by_cases hp_top : p = ∞
+  · have hq_top : q = ∞ := by rwa [hp_top, top_le_iff] at hpq
+    rw [hp_top]
+    rwa [hq_top] at hfq_lt_top
+  have hp_pos : 0 < p.toReal := ENNReal.toReal_pos hp0 hp_top
+  by_cases hq_top : q = ∞
+  · rw [snorm_eq_snorm' hp0 hp_top]
+    rw [hq_top, snorm_exponent_top] at hfq_lt_top
+    refine' lt_of_le_of_lt (snorm'_le_snormEssSup_mul_rpow_measure_univ hp_pos) _
+    refine' ENNReal.mul_lt_top hfq_lt_top.ne _
+    exact (ENNReal.rpow_lt_top_of_nonneg (by simp [hp_pos.le]) (measure_ne_top μ Set.univ)).ne
+  have hq0 : q ≠ 0 := by
+    by_contra hq_eq_zero
+    have hp_eq_zero : p = 0 := le_antisymm (by rwa [hq_eq_zero] at hpq) (zero_le _)
+    rw [hp_eq_zero, ENNReal.zero_toReal] at hp_pos
+    exact (lt_irrefl _) hp_pos
+  have hpq_real : p.toReal ≤ q.toReal := by rwa [ENNReal.toReal_le_toReal hp_top hq_top]
+  rw [snorm_eq_snorm' hp0 hp_top]
+  rw [snorm_eq_snorm' hq0 hq_top] at hfq_lt_top
+  exact snorm'_lt_top_of_snorm'_lt_top_of_exponent_le hfq_m hfq_lt_top (le_of_lt hp_pos) hpq_real
+#align measure_theory.mem_ℒp.mem_ℒp_of_exponent_le MeasureTheory.Memℒp.memℒp_of_exponent_le
+
+end SameSpace
+
+section Bilinear
+
+variable {α E F G : Type*} {m : MeasurableSpace α}
+  [NormedAddCommGroup E] [NormedAddCommGroup F] [NormedAddCommGroup G] {μ : Measure α}
+  {f : α → E} {g : α → F}
+
```

Comment:
Could you also move `snorm_le_snorm_top_mul_snorm` and `snorm_le_snorm_mul_snorm_top` around here? These are particular cases of the next inequalities when one exponent is infinite, so it would feel strange to separate them.